db/migrate/20150612135546_create_host_status.rb

```diff
@@ -33,4 +40,41 @@ def down
 
     drop_table :host_status
   end
+
+  private
+
+  def update_statuses(host)
+    results = HostStatus.status_registry.map do |status_class|
+      status = host.get_status(status_class)
+      update_sub_status(status)
+    end
+
+    if results.any?
+      host.host_statuses.reload
+      host.refresh_global_status
+
+      unless (saved = host.save(:validate => false))
+        logger.warn "Skipping global status refresh for host #{host.name}"
+      end
+      results.push saved
+    end
+
+    !results.include?(false)
```

Comment:
the warning message gets printed here https://github.com/theforeman/foreman/pull/3554/files#diff-4e168c0281109f0142eb253665fad464R22